### DISCUSSION TITLE
启动器不加 `-Dlog4j.configurationFile` 真的安全吗

### DISCUSSION THREAD

**Discussion by zkitefly** [Meloong-Git/PCL#2654] [提问] [answered]
我注意到启动器的启动脚本里面不会有 `-Dlog4j.configurationFile` 参数，这会不会造成些安全隐患呢

**Comment by Pigeon0v0** [comment]
HMCL 有这个是因为他们自己有 Log4j 的配置文件，PCL2 使用 NoLookup 关了，同时 Mojang 应该也官方修复了这个问题

**Reply by zkitefly** [reply]
我记得低版本 + Forge 用 NoLookup 封不死的？

**Comment by LTCatt** [comment, marked as answer]
Mojang 官方已经对 assetIndex 发布了热更新，没问题的 =。=
现在其实理论上启动器什么额外参数都不加也不会出现问题（

**Comment by zkitefly** [comment]
> Mojang 官方已经对 assetIndex 发布了热更新，没问题的 =。= 现在其实理论上启动器什么额外参数都不加也不会出现问题（

---

> 一方面是 forge 不会用这个东西，另一方面是它真的用了 json 里提供的 log4j2.xml 了吗？

> 启动参数里面只有 -Dlog4j2.formatMsgNoLookups=true，没有-Dlog4j.configurationFile

> 那就说明它不会使用旧版 Minecraft 后续更新过的 xml

-- 来自和他人的交流

那这说明旧版本可能会有漏洞么？

**Reply by LTCatt** [reply]
我不确定……mj 官启加了这些参数么
